lib/vanagon/platform/windows.rb

```diff
@@ -34,6 +37,30 @@ def package_name(project)
         "#{project.name}-#{project.version}-#{project.release}-#{@architecture}.msi"
       end
 
+      # Add a repository (or install Chocolatey)
+      # Note - this only prepares the list of commands to be executed once the Platform
+      # has been setup
+      #
+      # @param definition [String] Definition for adding repo, can be a Repo URL (including file:)
+      #    If file suffix is 'ps1' it is downloaded and executed to install chocolatey
+      # @return [Array]  Commands to executed after platform startup
+      def add_repository(definition)
+        definition = URI.parse(definition)
+        commands = []
+
+        if definition.scheme =~ /^(http|ftp|file)/
+          if File.extname(definition.path) == '.ps1'
+            commands << %Q{powershell.exe -NoProfile -ExecutionPolicy Bypass -Command 'iex ((new-object net.webclient).DownloadString(\"#{definition}\"))'}
+          else
+            commands << %Q{C:/ProgramData/chocolatey/bin/choco.exe source add -n #{definition.host}-#{definition.path.gsub('/', '-')} -s "#{definition}" --debug || echo "Oops, it seems that you don't have chocolatey installed on this system. Please ensure it's there by adding something like 'plat.add_repository 'https://chocolatey.org/install.ps1'' to your platform definition."}
```

Comment:
@melissa 
1. The gsub format above gives an untidy name, but it does work out as unique as each URL itself *should* be unique.
2. Its difficult to very the URL is an actual repo until we do the choco add source - but since we have error detection in the actual add source command, that will fail if the repo is invalid, so we kind of do have verification.